flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/CdcDataTypeTransformerTest.java

```diff
@@ -75,6 +76,68 @@ public void testCharTypeForPrimaryKey() {
+        new DecimalType(10, 0)
+                .accept(new StarRocksUtils.CdcDataTypeTransformer(true, bigIntLengthBuilder));
+        StarRocksColumn bigIntLengthColumn = bigIntLengthBuilder.build();
+        assertEquals("primary_key", bigIntLengthColumn.getColumnName());
+        assertEquals(0, bigIntLengthColumn.getOrdinalPosition());
+        assertEquals(StarRocksUtils.BIGINT, bigIntLengthColumn.getDataType());
+        assertTrue(bigIntLengthColumn.isNullable());
+
+        // map to LARGEINT of StarRocks if primary key column with precision < 18 and scale = 0
+        StarRocksColumn.Builder unsignedBigIntLengthBuilder =
+                new StarRocksColumn.Builder().setColumnName("primary_key").setOrdinalPosition(0);
+        new DecimalType(20, 0)
+                .accept(
+                        new StarRocksUtils.CdcDataTypeTransformer(
+                                true, unsignedBigIntLengthBuilder));
+        StarRocksColumn unsignedBigIntLengthColumn = unsignedBigIntLengthBuilder.build();
+        assertEquals("primary_key", unsignedBigIntLengthColumn.getColumnName());
+        assertEquals(0, unsignedBigIntLengthColumn.getOrdinalPosition());
+        assertEquals(StarRocksUtils.LARGEINT, unsignedBigIntLengthColumn.getDataType());
+        assertTrue(unsignedBigIntLengthColumn.isNullable());
+
+        // map to Varchar of StarRocks if primary key column with precision >= 38 and scale = 0
+        StarRocksColumn.Builder outOfBoundLengthBuilder =
+                new StarRocksColumn.Builder().setColumnName("primary_key").setOrdinalPosition(0);
+        new DecimalType(38, 0)
+                .accept(new StarRocksUtils.CdcDataTypeTransformer(true, outOfBoundLengthBuilder));
+        StarRocksColumn outOfBoundColumn = outOfBoundLengthBuilder.build();
+        assertEquals("primary_key", outOfBoundColumn.getColumnName());
+        assertEquals(0, outOfBoundColumn.getOrdinalPosition());
+        assertEquals(StarRocksUtils.VARCHAR, outOfBoundColumn.getDataType());
+        assertEquals(Integer.valueOf(38), outOfBoundColumn.getColumnSize().orElse(null));
+        assertTrue(unsignedBigIntLengthColumn.isNullable());
+
+        // map to Varchar of StarRocks if primary key column with scale > 0
+        StarRocksColumn.Builder scaleBuilder =
+                new StarRocksColumn.Builder().setColumnName("primary_key").setOrdinalPosition(0);
+        new DecimalType(12, 1)
+                .accept(new StarRocksUtils.CdcDataTypeTransformer(true, scaleBuilder));
+        StarRocksColumn scaleBoundColumn = scaleBuilder.build();
+        assertEquals("primary_key", scaleBoundColumn.getColumnName());
+        assertEquals(0, scaleBoundColumn.getOrdinalPosition());
+        assertEquals(StarRocksUtils.VARCHAR, scaleBoundColumn.getDataType());
+        assertEquals(Integer.valueOf(12), scaleBoundColumn.getColumnSize().orElse(null));
+        assertTrue(unsignedBigIntLengthColumn.isNullable());
```

Comment:
Minor suggestion: maybe create an ad-hoc function like `checkPrimaryColumnTypeCast(DataType in, DataType expected)` to reduce duplicated codes?